src/logging.c

```diff
@@ -31,6 +41,27 @@ void rcutils_logging_initialize()
   if (!g_rcutils_logging_initialized) {
     g_rcutils_logging_output_handler = &rcutils_logging_console_output_handler;
     g_rcutils_logging_severity_threshold = RCUTILS_LOG_SEVERITY_INFO;
+
+    // Check for the environment variable for custom output formatting
+    const char * output_format;
+    const char * ret = rcutils_get_env("RCUTILS_CONSOLE_OUTPUT_FORMAT", &output_format);
+    if (NULL == ret && strcmp(output_format, "") != 0) {
+      size_t chars_to_copy = strlen(output_format);
+      if (chars_to_copy > RCUTILS_LOGGING_MAX_OUTPUT_FORMAT_LEN - 1) {
+        chars_to_copy = RCUTILS_LOGGING_MAX_OUTPUT_FORMAT_LEN - 1;
+      }
+      memcpy(g_rcutils_logging_output_format_string, output_format, chars_to_copy);
+      g_rcutils_logging_output_format_string[chars_to_copy] = '\0';
+    } else {
+      if (NULL != ret) {
+        fprintf(
+          stderr,
+          "Failed to get output format from env. variable: %s. Using default output format.\n",
+          ret);
+      }
```

Comment:
It might make more sense to return early here on error, but since there isn't a way to signal the error, the logging calls will still be called and continually try to re-initialise the logging system. So, I chose to just fallback to the default format.